docs/elasticsearch/ux.md

```diff
@@ -0,0 +1,1081 @@
+---
+layout: default
+title: Search Experience
+parent: Elasticsearch
+nav_order: 12
+---
+
+# Search Experience
+
+Expectations from search engines have evolved over the years. Just returning relevant results quickly is no longer enough for most users. Elasticsearch includes many features that enhance the user’s search experience as follows:
+
+Feature | Description
+:--- | :---
+Autocomplete queries | Suggest phrases as the user types.
+Paginate results |  Rather than a single, long list, break search results into pages.
+Scroll search | Return a large number of results in batches.
+Sort results | Allow sorting results by different criteria.
+Highlight query matches | Highlight the search term in the results.
+
+---
+
+## Autocomplete queries
+
+Autocomplete shows suggestions to users while they type.
+
+For example, if a user types "pop," Elasticsearch provides suggestions like "popcorn" or "popsicles." These suggestions preempt your user's intention and lead them to a possible search term more quickly.
+
+Elasticsearch allows you to design autocomplete that’s:
+
+- Responsive: Updates suggestions with each keystroke.
+- Relevant: Serves a few but relevant suggestions.
+- Forgiving: Tolerates typos.
+
+Implement autocomplete in one of three methods:
+
+- Prefix matching
+- Edge N-gram matching
+- Completion suggesters
+
+These methods are described below.
+
+### Prefix matching
+
+Prefix matching finds documents that matches the last term in the query string.
+
+For example, assume that the user types “qui” into a search UI. To autocomplete this phrase, use the `match_phrase_prefix` query to search all `text_entry` fields that begin with the prefix "qui."
+To make the word order and relative positions flexible, specify a `slop` value. To learn about the `slop` option, see [Options](../full-text/#options).
+
+#### Sample Request
+
+```json
+GET shakespeare/_search
+{
+  "query": {
+    "match_phrase_prefix": {
+      "text_entry": {
+        "query": "qui",
+        "slop": 3
+      }
+    }
+  }
+}
+```
+
+Prefix matching doesn’t require any special mappings. It works with your data as-is.
+However, it’s a fairly resource-intensive operation. A prefix of `a` could match hundreds of thousands of terms and not be useful to your user.
+To limit the impact of prefix expansion, set `max_expansions` to a reasonable number. To learn about the `max_expansions` option, see [Options](../full-text/#options).
+
+#### Sample Request
+
+```json
+GET shakespeare/_search
+{
+  "query": {
+    "match_phrase_prefix": {
+      "text_entry": {
+        "query": "qui",
+        "slop": 3,
+        "max_expansions": 10
+      }
+    }
+  }
+}
+```
+
+The ease of implementing query-time autocomplete comes at the cost of performance.
+When implementing this feature on a large scale, we recommend an index-time solution. With an index-time solution, you might experience slower indexing, but it’s a price you pay only once and not for every query.
```

Comment:
Probably worth noting that the other two options are index-time.

Reply:
Done.